From beb16dad42dcba63ff3e7836c7ddd6498743e1ba Mon Sep 17 00:00:00 2001
From: Sun Haiyong <sunhaiyong@zdbr.net>
Date: Thu, 10 Aug 2023 03:48:15 +0000
Subject: [PATCH] fix amdgpu for wine.

Note from cth451: 3c5000 uses 16K paging and radeon would complain
when allocating 4K aligned pages for buffer.

---
 src/gallium/winsys/amdgpu/drm/amdgpu_bo.c     | 2 ++
 src/gallium/winsys/radeon/drm/radeon_drm_bo.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/gallium/winsys/amdgpu/drm/amdgpu_bo.c b/src/gallium/winsys/amdgpu/drm/amdgpu_bo.c
index be79533..acb0585 100644
--- a/src/gallium/winsys/amdgpu/drm/amdgpu_bo.c
+++ b/src/gallium/winsys/amdgpu/drm/amdgpu_bo.c
@@ -485,6 +485,8 @@ static struct amdgpu_winsys_bo *amdgpu_create_bo(struct amdgpu_winsys *ws,
       pb_cache_init_entry(&ws->bo_cache, bo->cache_entry, &bo->base,
                           heap);
    }
+   if(alignment < 16384)
+       alignment = 16384;
    request.alloc_size = size;
    request.phys_alignment = alignment;
 
diff --git a/src/gallium/winsys/radeon/drm/radeon_drm_bo.c b/src/gallium/winsys/radeon/drm/radeon_drm_bo.c
index 6ffeab7..b561495 100644
--- a/src/gallium/winsys/radeon/drm/radeon_drm_bo.c
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_bo.c
@@ -611,6 +611,8 @@ static struct radeon_bo *radeon_create_bo(struct radeon_drm_winsys *rws,
    assert((initial_domains &
            ~(RADEON_GEM_DOMAIN_GTT | RADEON_GEM_DOMAIN_VRAM)) == 0);
 
+   if(alignment < 16384)
+      alignment = 16384;
    args.size = size;
    args.alignment = alignment;
    args.initial_domain = initial_domains;
-- 
2.31.1

